src/common/components/Select/style.css.ts

```diff
@@ -0,0 +1,143 @@
+import { containerSize } from '@components/Input/constants';
+import { formColors } from '@constants/styleValues';
+import { style, styleVariants } from '@vanilla-extract/css';
+
+import { theme } from 'styles/theme.css';
+
+export const container = style({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: 9,
+
+  color: theme.color.black,
+  ...theme.font.BODY_1_18_M,
+});
+
+export const containerVar = styleVariants(containerSize, (size) => [container, { width: size }]);
+
+export const title = style({
+  display: 'flex',
+  alignItems: 'center',
+  gap: 6,
+
+  color: theme.color.baseText,
+  ...theme.font.TITLE_5_18_SB,
+});
+
+export const circle = style({
+  width: 8,
+  height: 8,
+  borderRadius: 4,
+  backgroundColor: theme.color.primary,
+});
+
+export const selectContainer = style({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: 9,
+  position: 'relative',
+});
+
+export const select = style({
+  display: 'flex',
+
+  flex: 1,
+
+  padding: 16,
+  borderRadius: 12,
+
+  ...theme.font.BODY_2_16_R,
+  color: theme.color.placeholder,
+
+  cursor: 'pointer',
+
+  zIndex: 1,
+
+  ':focus': {
+    boxShadow: `0 0 0 1px ${theme.color.primary} inset`,
+  },
+
+  ':disabled': {
+    backgroundColor: theme.color.subBackground, // gray30 -> 20으로 수정해야함
+    color: theme.color.lighterText,
+  },
+});
+
+export const selectVariant = styleVariants(formColors, (styles) => [
+  select,
+  {
+    ...styles,
+  },
+]);
+
+export const icon = style({
+  position: 'absolute',
+  top: 17,
+  right: 16,
+  width: 24,
+  color: theme.color.dropdownButton,
+
+  transition: 'transform 0.3s ease-out',
+
+  selectors: {
+    [`${select}:focus+&`]: {
+      transform: 'rotate(180deg)',
+    },
+    [`${select}:disabled+&`]: {
+      display: 'none',
+    },
+  },
+});
+
+export const optionContainer = style({
+  display: 'flex',
+  flexDirection: 'column',
+
+  position: 'absolute',
+  width: '100%',
+  top: 50,
+
+  padding: '15px 8px',
+
+  backgroundColor: theme.color.white,
+
+  borderRadius: 12,
+  boxShadow: `0 0 0 1px ${theme.color.border} inset`,
+
+  opacity: 0,
+  transition: '0.3s ease-in-out',
+
+  selectors: {
+    [`${select}:focus~&`]: {
+      opacity: 1,
+      transform: 'translateY(17px)',
+    },
```

Comment:
opacity 0만 주어서 사실은 렌더링이 되고 있는 상황인데요. 그래서 option이 있는 빈 공간 클릭하면 클릭이 되고 있습니다!

```suggestion
  opacity: 0,
  pointerEvents: 'none', // 마우스, 키보드 클릭 막기
  visibility: 'hidden', // screen readers에서 제외
  transition: '0.3s ease-in-out',

  selectors: {
    [`${select}:focus~&`]: {
      opacity: 1,
      pointerEvents: 'auto',
      visibility: 'visible',
      transform: 'translateY(17px)',
    },
```
위의 속성들도 추가해주어야 할 거 같아요~

Reply:
버그잡아주셔서 감사합니다
pointerEvents 이 친구는 저대로 넣으면 option을 클릭해도 selectBox에 반영이 안되더라고요? 
pointerEvents 를 제어하지 않아도, visibility 만으로도 클릭을 충분히 막을 수 있어서 visibility 속성만 추가해주어 해결했습니다! 

추가로 transform 애니메이션으로 인해 option 클릭 정확도 떨어지는게 너무 거슬려서 그냥 transform 빼주고 opacity 애니메이션만 남겨두었습니다.

Reply:
오홍! 그렇군요,,!!! 좋습니다 ◡̈